tooling/nargo_fmt/src/formatter/chunks.rs

```diff
@@ -0,0 +1,776 @@
+    PushIndentation,
+    /// Set the current indentation by popping it from the indentation stack.
+    PopIndentation,
+}
+
+impl Chunk {
+    pub(crate) fn width(&self) -> usize {
+        match self {
+            Chunk::Text(chunk) | Chunk::TrailingComment(chunk) | Chunk::LeadingComment(chunk) => {
+                chunk.width
+            }
+            Chunk::Group(group) => group.width(),
+            Chunk::SpaceOrLine => 1,
+            Chunk::Line { .. }
+            | Chunk::IncreaseIndentation
+            | Chunk::DecreaseIndentation
+            | Chunk::TrailingComma
+            | Chunk::PushIndentation
+            | Chunk::PopIndentation => 0,
+        }
+    }
+
+    /// Computes the width of this chunk considering it's inside an ExpressionList.
+    /// The only thing that changes here compared to `width` is that a LambdaAsLastExpressionInList's
+    /// width is considered to be only the first line, so we can avoid splitting the entire call
+    /// arguments into separate lins.
+    pub(crate) fn width_inside_an_expression_list(&self) -> usize {
+        if let Chunk::Group(group) = &self {
+            if let GroupKind::LambdaAsLastExpressionInList { first_line_width, .. } = &group.kind {
+                return *first_line_width;
+            }
+        }
+
+        self.width()
+    }
+
+    pub(crate) fn has_newlines(&self) -> bool {
+        match self {
+            Chunk::Text(chunk) | Chunk::TrailingComment(chunk) | Chunk::LeadingComment(chunk) => {
+                chunk.has_newlines
+            }
+            Chunk::Group(group) => group.has_newlines(),
+            Chunk::TrailingComma
+            | Chunk::Line { .. }
+            | Chunk::SpaceOrLine
+            | Chunk::IncreaseIndentation
+            | Chunk::DecreaseIndentation
+            | Chunk::PushIndentation
+            | Chunk::PopIndentation => false,
+        }
+    }
+
+    /// Returns the current chunk as a Group, if it is one. Otherwise returns None.
+    pub(crate) fn group(self) -> Option<ChunkGroup> {
+        if let Chunk::Group(group) = self {
+            Some(group)
+        } else {
+            None
+        }
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct ChunkGroup {
+    pub(crate) chunks: Vec<Chunk>,
+
+    /// If `true`, when formatting in multiple lines, and after a SpaceOrLine,
+    /// a line will be written.
+    /// If `false`, when formatting in multiple lines, and after a SpaceOrLine,
+    /// a space will be inserted and the next chunk will go in the same line if
+    /// it fits that line.
+    ///
+    /// This is used to, for example, control how arrays are formatted. If each
+    /// element is short, we'll format the array like this:
+    ///
+    /// [
+    ///   1, 2, 3,
+    ///   4, 5
+    /// ]
+    ///
+    /// but if one of the elements is long, each one will go in a separate line:
+    ///
+    /// [
+    ///     1,
+    ///     1234567890123,
+    ///     3
+    /// ]
```

Comment:
```suggestion
    /// ```
    /// [
    ///     1,
    ///     1234567890123,
    ///     3
    /// ]
    /// ```
```

Without the surrounding ``` it appears inline as a hyperlink, at least in the Rust Analyser tooltips.
Same goes for the other array.